MetaMask/metamask-extension
# fix: display native ticker and logo for current chain and render near-zero amounts in simulations

## **Description**

* near-zero amounts were being displayed as zero – we check if our rounding produces zero for non-zero values, in which case we display MIN_DISPLAY_AMOUNT.
* add support for non-ETH native tokens.
* added stories for small amount and non-ETH native token

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23711?quickstart=1)

## **Related issues**

Fixes: [#2295](https://github.com/MetaMask/MetaMask-planning/issues/2295) [#2296](https://github.com/MetaMask/MetaMask-planning/issues/2296) [#2316]( https://github.com/MetaMask/MetaMask-planning/issues/2316)

## **Manual testing steps**

Switch to a network where ETH is not the native token, e.g. Polygon or Optimism.
1. Prepare to send a very small amount of the native currency (like 0.000000001) to a random address (not your own wallet).
2. Before confirming you should see the amount `- <0.000001` in the native currency of the current chain. 

For polygon, that looks like:
![image](https://github.com/MetaMask/metamask-extension/assets/507015/be36947e-cd07-46f0-943c-25dcec35e0cc)


## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
